File: src/cpp/100.same-tree.cpp
#include <vector>
#include <queue>
#include "lc_types.h"

using namespace std;
/*
 * @lc app=leetcode id=100 lang=cpp
 *
 * [100] Same Tree
 */

// @lc code=start
#ifndef LC_LOCAL_TEST
class Solution {
public:
#endif

    bool isSameTree(TreeNode* p, TreeNode* q) {
        if (!p && !q) return true;
        else if (!p || !q) return false;

        return ((p->val == q->val)
                 && isSameTree(p->left, q->left) 
                 && isSameTree(p->right, q->right));
    }

#ifndef LC_LOCAL_TEST
};
#endif
// @lc code=end

File: src/java/tree/100.same-tree.java
class P100 {
    // @lc app=leetcode id=100 lang=java
    // @lc code=start
    public class Solution {
        public boolean isSameTree(TreeNode p, TreeNode q) {
            if (p==null || q==null) return p==q;
            if (p.val!=q.val) return false;
            return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
        }
    }
    // @lc code=end
}
